src/AaveCurator/AaveCurator.sol

```diff
@@ -0,0 +1,301 @@
+    IWstEth(AaveV3EthereumAssets.wstETH_UNDERLYING).wrap(amount);
+
+    IERC20(AaveV3EthereumAssets.wstETH_UNDERLYING).safeTransfer(
+      address(AaveV3Ethereum.COLLECTOR),
+      IERC20(AaveV3EthereumAssets.wstETH_UNDERLYING).balanceOf(address(this))
+    );
+  }
+
+  /// @notice Withdraws wETH into ETH, then deposits for rETH
+  function depositRETH(uint256 amount) external onlyOwnerOrManager {
+    WETH.withdraw(amount);
+    ROCKET_POOL.deposit{value: amount}();
+
+    IERC20(AaveV3EthereumAssets.rETH_UNDERLYING).safeTransfer(
+      address(AaveV3Ethereum.COLLECTOR),
+      IERC20(AaveV3EthereumAssets.rETH_UNDERLYING).balanceOf(address(this))
+    );
+  }
+
+  function depositTokenIntoV2(address token, uint256 amount) external onlyOwnerOrManager {
+    if (!allowedFromTokens[token]) revert InvalidToken();
+    IERC20(token).approve(address(AaveV2Ethereum.POOL), amount);
+    AaveV2Ethereum.POOL.deposit(token, amount, address(AaveV3Ethereum.COLLECTOR), 0);
+    emit DepositedIntoV2(token, amount);
+  }
+
+  function depositTokenIntoV3(address token, uint256 amount) external onlyOwnerOrManager {
+    if (!allowedFromTokens[token]) revert InvalidToken();
+    IERC20(token).approve(address(AaveV3Ethereum.POOL), amount);
+    AaveV3Ethereum.POOL.deposit(token, amount, address(AaveV3Ethereum.COLLECTOR), 0);
+    emit DepositedIntoV3(token, amount);
+  }
+
+  function setManager(address _manager) external onlyOwner {
+    if (_manager == address(0)) revert Invalid0xAddress();
+    address oldManager = manager;
+    manager = _manager;
+    emit ManagerChanged(oldManager, manager);
+  }
+
+  function removeManager() external onlyOwner {
+    address oldManager = manager;
+    manager = address(0);
+    emit ManagerChanged(oldManager, manager);
+  }
+
+  function setAllowedFromToken(
+    address token,
+    address oracle,
+    bool isEthBased,
+    bool allowed
+  ) external onlyOwner {
+    if (token == address(0)) revert Invalid0xAddress();
+    if (oracle == address(0)) revert Invalid0xAddress();
+    allowedFromTokens[token] = allowed;
+    tokenChainlinkOracle[token] = TokenChailinkOracleData(oracle, isEthBased);
+    emit TokenUpdated(token, allowed);
+  }
+
+  function setAllowedToToken(
+    address token,
+    address oracle,
+    bool isEthBased,
+    bool allowed
+  ) external onlyOwner {
+    if (token == address(0)) revert Invalid0xAddress();
+    if (oracle == address(0)) revert Invalid0xAddress();
+    allowedToTokens[token] = allowed;
+    tokenChainlinkOracle[token] = TokenChailinkOracleData(oracle, isEthBased);
+    emit TokenUpdated(token, allowed);
+  }
+
+  function setMilkmanAddress(address _milkman) external onlyOwner {
+    if (_milkman == address(0)) revert Invalid0xAddress();
+    milkman = _milkman;
+  }
+
+  /// @notice Transfer any tokens on this contract to Aave V3 Collector
+  /// @param tokens List of token addresses
+  function withdrawToCollector(address[] calldata tokens) external onlyOwnerOrManager {
```

Comment:
one could consider using https://github.com/bgd-labs/solidity-utils/blob/main/src/contracts/utils/Rescuable.sol#L13 instead.
What is the point of the "manager" withdrawing?

Reply:
The point of the manager withdrawing is if funds are kept on curator to be swapped at different times, and to tie it all together with the StrategicAssetsManager as well. Maybe it can withdraw to collector or allowed address instead.